### PR TITLE
Added the BRS to Set Min TLS version for Event Hub Namespace

### PR DESCRIPTION
**Pull request by jaiswalricha**
Added the BRS to Set Min TLS version for Event Hub Namespace

### CODE REVIEW

**Review by aakashpahwa-msft — CHANGES_REQUESTED** [review]

File: Control coverage/Feature/EventHub.md [hunk 1]
```diff
@@ -125,3 +126,53 @@ api-version=2017-04-01<br />
...
+### Control Settings 
+
+
+### Control Spec 
+
+> **Passed:** 
+> TLS version of Event Hub Namespace is already defined as per the Security Recommendation.
+> 
+> **Failed:** 
+> Current minimum TLS version is less than required secured version .
```
[the remark this answers is not in the document]
Comment (jaiswalricha) [reply Control coverage/Feature/EventHub.md:146]: done.

File: Control coverage/Feature/EventHub.md [hunk 2]
```diff
@@ -125,3 +126,53 @@ api-version=2017-04-01<br />
...
+     Set-MinTLSVersionForEventHubNamespace
+
+     # For more help run:
+	 Get-Help Set-MinTLSVersionForEventHubNamespace -full
+	 ```  
+
+### Azure Policy or ARM API used for evaluation 
+
+- ARM API to list all Authorization Rules for an Event Hubs Namespace: /subscriptions/{0}/providers/Microsoft.EventHub/namespaces?api-version=2022-01-01-preview<br />
+**Properties:** properties.rights
```
[the remark this answers is not in the document]
Comment (jaiswalricha) [reply Control coverage/Feature/EventHub.md:172]: done

File: Scripts/RemediationScripts/AutoRemediation.ps1 [hunk 3]
```diff
@@ -162,6 +162,7 @@ Enter the choice (1|2)";
                 ($control.ControlId -eq "Azure_DBForMySQLFlexibleServer_DP_Use_Secure_TLS_Version_Trial") -or
                 ($control.ControlId -eq "Azure_SQLDatabase_DP_Enable_TDE") -or
                 ($control.ControlId -eq "Azure_FrontDoor_DP_Use_Secure_TLS_Version_Trial")-or
+                ($control.ControlId -eq "Azure_FrontDoor_DP_Use_Secure_TLS_Version_Trial")-or
```
[the remark this answers is not in the document]
Comment (jaiswalricha) [reply Scripts/RemediationScripts/AutoRemediation.ps1:165]: ahh, went wrong while merging, thanks for highlighting, corrected now.

File: Scripts/RemediationScripts/Remediate-SetEventHubNamespaceMinTLSVersion.ps1 [hunk 4]
```diff
@@ -0,0 +1,856 @@
+﻿<###
+# Overview:
+    This script is used to set Tls version for Event Hub namespaces in a Subscription.
```
[the remark this answers is not in the document]
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-SetEventHubNamespaceMinTLSVersion.ps1:3]: corrected it.

File: Scripts/RemediationScripts/Remediate-SetEventHubNamespaceMinTLSVersion.ps1 [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+            }
+            catch
+            {
+                $EventHubNamespaceSkipped += $EventHubNamespace
+                $logResource = @{}	
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                $logResource.Add("ResourceName",($_.ResourceName))	
+                $logResource.Add("Reason","Error configuring TLS version for Event Hub Namespace")    	
+                $logSkippedResources += $logResource	
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
```
[the remark this answers is not in the document]
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-SetEventHubNamespaceMinTLSVersion.ps1]: done

File: Scripts/RemediationScripts/ControlRemediationSpecification.json [hunk 6]
```diff
@@ -194,17 +194,28 @@
...
+        }
+      },
+      {
+        "ControlId": "Azure_EventHub_DP_Use_Secure_TLS_Version",
+        "EnableRemediation": "true",
+        "RemediationScriptUrl": "https://raw.githubusercontent.com/azsk/AzTS-docs/main/Scripts/RemediationScripts/Remediate-SetEventHubNamespaceMinTLSVersion.ps1",
+        "LoadCommand": "Remediate-SetEventHubNamespaceMinTLSVersion",
+        "InitCommand": "Set-MinTLSVersionForEventHubNamespace",
+        "RollbackMetadata": {
+          "RollbackCommand": "Set-PreviousTLSVersionForEventHubNamespace",
```
[the remark this answers is not in the document]
Comment (jaiswalricha) [reply Scripts/RemediationScripts/ControlRemediationSpecification.json]: corrected it to reset-function to keep it in sync

**Review by aakashpahwa-msft — APPROVED** [review]
LGTM

File: 08-Secure by default/Secure Templates/Bicep/EventHub.bicep [hunk 7]
```diff
@@ -8,13 +8,16 @@ param eventHubName string
...
-resource eventHubNamespace_resource 'Microsoft.EventHub/namespaces@2017-04-01' = {
+resource eventHubNamespace_resource 'Microsoft.EventHub/namespaces@2022-01-01-preview' = {
   sku: {
     name: 'Standard'
     tier: 'Standard'
     capacity: 1
   }
   name: eventHubNamespace
+  properties: {
+       minimumTlsVersion: '1.2'
```
Comment (Aboli-msft) [inline 08-Secure by default/Secure Templates/Bicep/EventHub.bicep:19]: Please add Conrol Name here too

Comment (jaiswalricha) [reply 08-Secure by default/Secure Templates/Bicep/EventHub.bicep]: added the control id documentation here.

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**